x/sequencer/keeper/sequencer.go

```diff
@@ -37,10 +37,22 @@ func (k Keeper) UpdateSequencer(ctx sdk.Context, sequencer types.Sequencer, oldS
 	seqByrollappKey := types.SequencerByRollappByStatusKey(sequencer.RollappId, sequencer.SequencerAddress, sequencer.Status)
 	store.Set(seqByrollappKey, b)
 
-	//handle unbonding queue changes due to status change
 	if sequencer.Status == oldStatus {
 		return
 	}
+
+	//rotate proposer if needed
+	if oldStatus == types.Proposer {
```

Comment:
dont think it's the cleanest to have this recursive logic here as this function should update the sequencer and it's doing more like rotating.
I'd keep this function clean and just update sequencer status and other implications of updating the sequencer 
 (like rotation etc) would put outside this function.

Reply:
gonna refactor